### DISCUSSION TITLE
[Feature Request]: Add a hideValueIf option for Column.

### DISCUSSION THREAD

**Discussion by amshehzad** [rappasoft/laravel-livewire-tables#1483] [Feature Requests]
### Overview

I think it'll be nice to pass the current row to `hideIf` so we can hide the column based on the row value e.g if a row is created_by admin, hide it

**Comment by lrljoe** [comment]
as discussed, suggest a hideValueIf approach

**Comment by lrljoe** [comment]
Assigned to @amshehzad initially, but can review

**Comment by lrljoe** [comment]
Should add the method hideValueIf() with a callback:
```php
hideValueIf(callback $callback)
```

The end code should add the usual functions has/get/set for the hideValueIf

If the hideValueIf callback returns false, the value should be returned
If the hideValueIf callback returns true, the value should not be returned

This should not use classes to hide the content, but actually remove the content, and return an empty <td></td>

**Comment by lrljoe** [comment]
In the meantime, this approach would work, in this example, it'll hide a "name" that is "Test123".  You could also push the row/value to an external method, and use that to return the value to display.

```php
  Column::make('Name', 'name')
          ->format(fn($value, $row, Column $column) => ($row->name == 'Test123' ? '' : $row->name)),
```

**Comment by leesherwood** [comment]
For anyone stumbling across this that needs a permission-based LinkColumn as I did, you can do the same as @lrljoe suggested using the `title` method.

```
LinkColumn::make('Action')
    ->title(fn($row) => auth()->user()->can('update', $row) ? 'Manage' : '')
    ->location(fn($row) => route('admin.users.edit', $row))
```

Yes, there's still an `<a>` in the source, but it's non-visible or clickable as it's 0px (with the default styling at least).

**Reply by lrljoe** [reply]
> For anyone stumbling across this that needs a permission-based LinkColumn as I did, you can do the same as @lrljoe suggested using the `title` method.
> 
> ```
> LinkColumn::make('Action')
>     ->title(fn($row) => auth()->user()->can('update', $row) ? 'Manage' : '')
>     ->location(fn($row) => route('admin.users.edit', $row))
> ```
> 
> Yes, there's still an `<a>` in the source, but it's non-visible or clickable as it's 0px (with the default styling at least).

100% - although for sanities sake on this, and ensuring that a user with a screenreader doesn't pick up the route (although you should be validating access on entry to the route), you may also want to use the same callback on the location:
```
LinkColumn::make('Action')
   ->title(fn($row) => auth()->user()->can('update', $row) ? 'Manage' : '')
   ->location(fn($row) => auth()->user()->can('update', $row) ? route('admin.users.edit', $row) : route('dashboard'))
 ```

The core issue when it comes to hiding a Column on a per-row basis, is:
- The Header will need to know if there's any rows that have a "Visible Column".
- Each row needs to know
   - Should it display the Column in full?
   - DO any other rows have the Column visible (thus should it just use an empty Column)
 
So displaying an empty value (as explained above) is the best way.

For things like Edit etc, I tend to use a Column with a custom view, which has all of the actions that a User would need to take.  This Column then displays the relevant Icons.
For example my Actions Column might present one or more of the following:
- Edit
- View
- Delete
- Create/View Notes
- Edit Permissions
- View Activity History

Which options are displayed, is calculated on a per-user, per-row basis, via an extended Column.

This is something I want to implement as a proper Column eventually.
